homeassistant/components/go2rtc/__init__.py

```diff
@@ -143,14 +147,17 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     # Validate the server URL
     try:
         client = Go2RtcRestClient(async_get_clientsession(hass), url)
-        await client.streams.list()
+        await client.validate_server_version()
     except Go2RtcClientError as err:
         if isinstance(err.__cause__, _RETRYABLE_ERRORS):
             raise ConfigEntryNotReady(
                 f"Could not connect to go2rtc instance on {url}"
             ) from err
         _LOGGER.warning("Could not connect to go2rtc instance on %s (%s)", url, err)
         return False
+    except Go2RtcVersionError as err:
+        _LOGGER.warning("The go2rtc server version is not supported, %s", err)
+        return False
```

Comment:
Not that go2rtc is a config entry we should raise instead

Reply:
If the user has control of the server, the user can update the server to run the correct version. Then we want to try to connect again by raising `ConfigEntryNotReady`.

Reply:
OK, that makes sense